Issue with TypedDict in new 2.13.1

**Describe the bug**
When using `TypedDict` I get the error:
```
        for argname, expected_type in memo.type_hints.items():
            if argname != 'return' and argname in memo.arguments:
                value = memo.arguments[argname]
                description = 'argument "{}"'.format(argname)
                try:
                    check_type(description, value, expected_type, memo)
                except TypeError as exc:  # suppress unnecessarily long tracebacks
>                   raise TypeError(*exc.args) from None
E                   TypeError: TypedDict does not support instance and class checks

.../lib/python3.8/site-packages/typeguard/__init__.py:874: TypeError
```
This seems to happen in 2.13.1 and 2.13.2 but not in 2.13.0.

--- a/docs/versionhistory.rst
+++ b/docs/versionhistory.rst
@@ -3,6 +3,11 @@ Version history
 
 This library adheres to `Semantic Versioning 2.0 <https://semver.org/#semantic-versioning-200>`_.
 
+**UNRELEASED**
+
+- Fixed ``TypedDict`` causing ``TypeError: TypedDict does not support instance and class checks``
+  on Python 3.8 with standard library (not ``typing_extensions``) typed dicts
+
 **2.13.2** (2021-11-23)
 
 - Fixed ``typing_extensions`` being imported unconditionally on Python < 3.9

--- a/src/typeguard/__init__.py
+++ b/src/typeguard/__init__.py
@@ -72,11 +72,15 @@ else:
     try:
         from typing_extensions import _TypedDictMeta
     except ImportError:
-        def is_typeddict(tp) -> bool:
-            return False
-    else:
-        def is_typeddict(tp) -> bool:
-            return isinstance(tp, _TypedDictMeta)
+        _TypedDictMeta = None
+
+    def is_typeddict(tp) -> bool:
+        # Check for the typing_extensions version first
+        if _TypedDictMeta is not None:
+            if isinstance(tp, _TypedDictMeta):
+                return True
+
+        return issubclass(tp, dict) and hasattr(tp, '__annotations__')
 
 
 if TYPE_CHECKING:

--- a/src/typeguard/__init__.py
+++ b/src/typeguard/__init__.py
@@ -63,24 +63,20 @@ except ImportError:
 
 if sys.version_info >= (3, 10):
     from typing import is_typeddict
-elif sys.version_info >= (3, 9):
-    def is_typeddict(tp) -> bool:
-        from typing import _TypedDictMeta
-
-        return isinstance(tp, _TypedDictMeta)
 else:
+    _typed_dict_meta_types = ()
+    if sys.version_info >= (3, 8):
+        from typing import _TypedDictMeta
+        _typed_dict_meta_types += (_TypedDictMeta,)
+
     try:
         from typing_extensions import _TypedDictMeta
+        _typed_dict_meta_types += (_TypedDictMeta,)
     except ImportError:
-        _TypedDictMeta = None
+        pass
 
     def is_typeddict(tp) -> bool:
-        # Check for the typing_extensions version first
-        if _TypedDictMeta is not None:
-            if isinstance(tp, _TypedDictMeta):
-                return True
-
-        return issubclass(tp, dict) and hasattr(tp, '__annotations__')
+        return isinstance(tp, _typed_dict_meta_types)
 
 
 if TYPE_CHECKING:

--- a/tests/test_typeguard_py36.py
+++ b/tests/test_typeguard_py36.py
@@ -1,3 +1,4 @@
+import sys
 import warnings
 from typing import Any, AsyncGenerator, AsyncIterable, AsyncIterator, Callable, Dict
 
@@ -7,9 +8,9 @@ from typing_extensions import Protocol, runtime_checkable
 from typeguard import TypeChecker, typechecked
 
 try:
-    from typing_extensions import TypedDict
-except ImportError:
     from typing import TypedDict
+except ImportError:
+    from typing_extensions import TypedDict
 
 
 @runtime_checkable
@@ -110,7 +111,9 @@ class TestTypeChecked:
             pass
 
         foo({'x': 1})
-        pytest.raises(TypeError, foo, {'y': 1})
+        if sys.version_info[:2] != (3, 8):
+            # TypedDict is unusable for runtime checking on Python 3.8
+            pytest.raises(TypeError, foo, {'y': 1})
 
     def test_mapping_is_not_typeddict(self):
         """Regression test for #216."""

